rust-nostd-avr/Dockerfile

```diff
@@ -0,0 +1,60 @@
+# it pulles latest nightly a base image, but because of LLVM bug installes a specifc nightly in the generate step
+# -> if resolved can used latest / new stable
+# https://github.com/rust-lang/compiler-builtins/issues/523
+# we could directly pull that special nightly version but the following problem: 
+# offical rust container dont have taged container :/
+# So two options.
+# 1) Use offical nightly rust-lang container and let it install the needed nightly version
+# 2) Use instrumentist container - would shrink the image a bit?
+# 
+# I think it would be save to use instrumentist container as they are just retagged versions
+# can check here -> 
+# https://github.com/rust-lang/docker-rust-nightly/issues/3
+# https://github.com/instrumentisto/rust-docker-image/blob/5a7de506e39bd8ce6ccda3d55830d333d764b874/nightly/buster/hooks/post_push#L26-L27
+# but i opted for the normal latest nightly container currently
+
+#FROM instrumentisto/rust:nightly-bullseye-2022-12-13
+FROM rustlang/rust:nightly-bullseye
+
+# Install avr-gcc, binutils and libc - used for linking + creating elf
+RUN apt-get update && apt-get install -y avr-libc binutils-avr gcc-avr
+
+RUN useradd -rm -d /home/wokwi wokwi
+USER wokwi
+WORKDIR /home/wokwi
+ENV USER=wokwi
+
+# Check rust version
+RUN cargo --version
+RUN rustup show
+
+# Install extra crates
+# it would be nice if we could get cargo-audit and sccache in some binary form :D this build step take's ages
+RUN cargo install cargo-audit cargo-generate sccache
```

Comment:
You could build another docker image that precompiled them, publish it on Dockerhub, and then copy it from that image. e.g.

```
COPY --from=user/image /path/to/sccache /path/to/sccache
```